vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java

```diff
@@ -49,12 +49,17 @@
     @Autowired
     private SequenceHistory sequenceHistory;
 
-    @RequestMapping(value = "/exhibit/space/{id}")
-    public String space(@PathVariable("id") String id, Model model) {
+    @RequestMapping(value = { "/exhibit/space/{id}", "/preview/{previewId}/space/{id}" })
+    public String space(@PathVariable("id") String id, @PathVariable(name = "previewId", required = false) String previewId, Model model) {
+        if (previewId != null) {
+            model.addAttribute("isExhPreview", true);
+            model.addAttribute("previewId", previewId);
+        }
```

Comment:
since that needs to be added in each controller, I would just move that into the aspect.